command/server.go

```diff
@@ -363,6 +365,19 @@ func (c *ServerCommand) Run(args []string) int {
 		c.logger = logging.NewVaultLoggerWithWriter(c.logWriter, level)
 	}
 
+	{
+		signalCh := make(chan os.Signal, 1)
+		signal.Notify(signalCh, syscall.SIGUSR2)
+		go func() {
+			var buf = make([]byte, 65536)
```

Comment:
This seems extremely small to me given panics we've seen from live running systems. I'd suggest it needs to be 1MB at the very minimum and should possibly be significantly more, up to 32 or 64MB.

Reply:
Fair enough.  How would you feel about a new config option?  Since we recommend disabling swap I'm reluctant to pre-allocate enough to handle the biggest systems, and then foist that on the smaller ones.

Reply:
I don't really see a one-time allocation of somewhere between 1 and say 32MB that can be GC'd almost immediately and is a result of a direct user action as a real burden. Especially since this will almost never be used outside of a direct debugging context with our team.

I forgot you're doing the allocation outside of the signal handler. Just put it inside the for loop and you don't have to have it preallocated if it's not being used.

Reply:
I put it outside the for loop thinking that if the user is in a situation where they need to look at stack traces, there might not be memory available to allocate.

Reply:
If they're in a state where there is no memory and no swap available they're probably edging on OOM anyways so at that point they can probably SIGQUIT :-)

Reply:
Ok.  Do we want to be clever in that when we observe the buf is fully used by the call to Stack(), assume it wasn't big enough, reallocate, and try again?

Reply:
Why?

Reply:
In case 64MB or whatever isn't enough.

Reply:
I guess if you want. I'd be shocked if 64MB isn't enough. I think in the majority of cases 1MB will be plenty. I think 32MB is a reasonable upper limit in all cases.

Reply:
Ok, let's start with 64MB then and we'll see whether it's an issue in practice before adding complexity.

Reply:
64 is fine, personally I'd go 32.